src/node.cpp

```diff
@@ -1096,11 +1102,13 @@ void Node::setpubliclink(handle ph, m_time_t cts, m_time_t ets, bool takendown)
     }
     else            // update
     {
+        client->mPublicLinks.erase(plink->ph);
```

Comment:
I think you don't need to `erase`. The new assignment a few lines below will do its work anyway. Instead, you could set a:
```suggestion
        assert(client->mPublicLinks.find(plink->ph) != client->mPublicLinks.end());
```
And for the block above to create the link:
```
        assert(client->mPublicLinks.find(ph) == client->mPublicLinks.end());
```

Reply:
I added the erase because in else we update the current plink->ph with ph received as param.
**plink->ph = ph;** so if ph has changed we need to remove current map entry and add new one.

However I've testing and I'm not sure ph can change so in this case this line **plink->ph = ph;** is not needed.

Reply:
I see. I assumed at some point in my head that the key of the map would be the nodehandle. I think we should adjust the map. It also helps here, since the item in the map only needs to update the value, rather than remove the item to be inserted again.